cmake/compiler/clang/compiler_flags.cmake

```diff
@@ -23,6 +23,9 @@ set_compiler_property(PROPERTY diagnostic -fcolor-diagnostics)
 # clang flag to save temporary object files
 set_compiler_property(PROPERTY save_temps -save-temps)
 
+# clang doesn't handle the -T flag
+set_compiler_property(PROPERTY linker_script -Wl,-T)
```

Comment:
let's have the `-Wl,-T` as the default ld linker script setting here (which can the be overruled by the gcc ld front-end):
https://github.com/zephyrproject-rtos/zephyr/blob/main/cmake/linker/ld/linker_flags.cmake

as:
```
set_property(TARGET linker PROPERTY linker_script "${LINKERFLAGPREFIX},-T")
```

Reply:
I think that's what I did -- the issue here is that clang starts with the gcc settings and modifies them, so I had to reset the property back to the default value.